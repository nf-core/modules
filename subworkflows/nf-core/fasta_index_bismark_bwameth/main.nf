include { UNTAR                     } from '../../../modules/nf-core/untar/main'
include { GUNZIP                    } from '../../../modules/nf-core/gunzip/main'
include { BISMARK_GENOMEPREPARATION } from '../../../modules/nf-core/bismark/genomepreparation/main'
include { BWAMETH_INDEX             } from '../../../modules/nf-core/bwameth/index/main'
include { SAMTOOLS_FAIDX            } from '../../../modules/nf-core/samtools/faidx/main'

workflow FASTA_INDEX_BISMARK_BWAMETH {

    take:
    fasta         // channel: [ val(meta), [ fasta ] ]
    fasta_index   // channel: [ val(meta), [ fasta index ] ]
    bismark_index // channel: [ val(meta), [ bismark index ] ]
    bwameth_index // channel: [ val(meta), [ bwameth index ] ]

    main:

    ch_fasta         = Channel.empty()
    ch_fasta_index   = Channel.empty()
    ch_bismark_index = Channel.empty()
    ch_bwameth_index = Channel.empty()
    ch_versions      = Channel.empty()

    if (fasta.toString().endsWith('.gz')) {
        GUNZIP (
            [ [:], file(fasta, checkIfExists: true) ]
        )
        ch_fasta    = GUNZIP.out.gunzip
        ch_versions = ch_versions.mix(GUNZIP.out.versions)
    } else {
        ch_fasta    = Channel.value([[:], file(fasta, checkIfExists: true)])
    }

    // Aligner: bismark or bismark_hisat
    if( params.aligner =~ /bismark/ ){
        /*
         * Generate bismark index if not supplied
         */
        if (bismark_index) {
            if (bismark_index.toString().endsWith('.gz')) {
                UNTAR (
                    [ [:], file(bismark_index, checkIfExists: true) ]
                )
                ch_bismark_index = UNTAR.out.untar
                ch_versions      = ch_versions.mix(UNTAR.out.versions)
            } else {
                ch_bismark_index = Channel.value([[:], file(bismark_index, checkIfExists: true)])
            }
        } else {
            BISMARK_GENOMEPREPARATION (
                ch_fasta
            )
            ch_bismark_index = BISMARK_GENOMEPREPARATION.out.index
            ch_versions      = ch_versions.mix(BISMARK_GENOMEPREPARATION.out.versions)
        }
    }

    // Aligner: bwameth
    else if ( params.aligner == 'bwameth' ){
        /*
         * Generate bwameth index if not supplied
         */
        if (bwameth_index) {
            if (bwameth_index.toString().endsWith('.gz')) {
                UNTAR (
                    [ [:], file(bwameth_index, checkIfExists: true) ]
                )
                ch_bwameth_index = UNTAR.out.untar
                ch_versions      = ch_versions.mix(UNTAR.out.versions)
            } else {
                ch_bwameth_index = Channel.value([[:], file(bwameth_index, checkIfExists: true)])
            }
        } else {
            BWAMETH_INDEX (
                ch_fasta
            )
            ch_bwameth_index = BWAMETH_INDEX.out.index
            ch_versions      = ch_versions.mix(BWAMETH_INDEX.out.versions)
        }

        /*
         * Generate fasta index if not supplied
         */
        if (fasta_index) {
            ch_fasta_index = Channel.value(file(fasta_index, checkIfExists: true))
        } else {
            SAMTOOLS_FAIDX(
                ch_fasta,
                [[:], []]
            )
            ch_fasta_index = SAMTOOLS_FAIDX.out.fai
            ch_versions    = ch_versions.mix(SAMTOOLS_FAIDX.out.versions)
        }
    }

    emit:
    fasta         = ch_fasta         // channel: [ val(meta), [ fasta ] ]
    fasta_index   = ch_fasta_index   // channel: [ val(meta), [ fasta index ] ]
    bismark_index = ch_bismark_index // channel: [ val(meta), [ bismark index ] ]
    bwameth_index = ch_bwameth_index // channel: [ val(meta), [ bwameth index ] ]
    versions      = ch_versions      // channel: [ versions.yml ]
}

