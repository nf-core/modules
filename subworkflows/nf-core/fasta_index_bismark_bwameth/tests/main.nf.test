nextflow_workflow {

    name "Test Subworkflow FASTA_INDEX_BISMARK_BWAMETH"
    script "../main.nf"
    workflow "FASTA_INDEX_BISMARK_BWAMETH"
    config "./nextflow.config"

    tag "subworkflows"
    tag "subworkflows_nfcore"
    tag "subworkflows/fasta_index_bismark_bwameth"
    tag "untar"
    tag "gunzip"
    tag "bismark/genomepreparation"
    tag "bwameth/index"
    tag "samtools/faidx"

    test("Params: bismark | download bismark index (bowtie2)") {

        when {
            params {
                aligner = "bismark"
            }
            workflow {
                """
                input[0] = file('https://github.com/nf-core/test-datasets/raw/methylseq/reference/genome.fa.gz', checkIfExists: true)
                input[1] = [] // fasta_index
                input[2] = file('https://github.com/nf-core/test-datasets/raw/methylseq/reference/Bowtie2_Index.tar.gz', checkIfExists: true)
                input[3] = [] // bwameth_index
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success},
                { assert snapshot(
                    workflow.out.fasta,
                    workflow.out.fasta_index,
                    workflow.out.bismark_index,
                    workflow.out.bwameth_index,
                    workflow.out.versions
                    ).match() }
            )
        }
    }

    test("Params: bismark | generate bismark index (bowtie2)") {

        when {
            params {
                aligner = "bismark"
            }
            workflow {
                """
                input[0] = file('https://github.com/nf-core/test-datasets/raw/methylseq/reference/genome.fa.gz', checkIfExists: true)
                input[1] = [] // fasta_index
                input[2] = [] // bismark_index
                input[3] = [] // bwameth_index
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success},
                { assert snapshot(
                    workflow.out.fasta,
                    workflow.out.fasta_index,
                    workflow.out.bismark_index,
                    workflow.out.bwameth_index,
                    workflow.out.versions
                    ).match() }
            )
        }
    }

    test("Params: bismark_hisat | download bismark index (hisat2)") {

        when {
            params {
                aligner = "bismark_hisat"
            }
            workflow {
                """
                input[0] = file('https://github.com/nf-core/test-datasets/raw/methylseq/reference/genome.fa.gz', checkIfExists: true)
                input[1] = [] // fasta_index
                input[2] = file('https://github.com/nf-core/test-datasets/raw/methylseq/reference/Hisat2_Index.tar.gz', checkIfExists: true)
                input[3] = [] // bwameth_index
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success},
                { assert snapshot(
                    workflow.out.fasta,
                    workflow.out.fasta_index,
                    workflow.out.bismark_index,
                    workflow.out.bwameth_index,
                    workflow.out.versions
                    ).match() }
            )
        }
    }

    test("Params: bismark_hisat | generate bismark index (hisat2)") {

        when {
            params {
                aligner = "bismark_hisat"
            }
            workflow {
                """
                input[0] = file('https://github.com/nf-core/test-datasets/raw/methylseq/reference/genome.fa.gz', checkIfExists: true)
                input[1] = [] // fasta_index
                input[2] = file('https://github.com/nf-core/test-datasets/raw/methylseq/reference/Hisat2_Index.tar.gz', checkIfExists: true)
                input[3] = [] // bwameth_index
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success},
                { assert snapshot(
                    workflow.out.fasta,
                    workflow.out.fasta_index,
                    workflow.out.bismark_index,
                    workflow.out.bwameth_index,
                    workflow.out.versions
                    ).match() }
            )
        }
    }

    test("Params: bwameth | download fasta index | download bwameth index") {

        when {
            params {
                aligner = "bwameth"
            }
            workflow {
                """
                input[0] = file('https://github.com/nf-core/test-datasets/raw/methylseq/reference/genome.fa.gz', checkIfExists: true)
                input[1] = file('https://github.com/nf-core/test-datasets/raw/methylseq/reference/genome.fa.fai', checkIfExists: true)
                input[2] = [] // bismark index
                input[3] = file('https://github.com/nf-core/test-datasets/raw/methylseq/reference/Bwameth_Index.tar.gz', checkIfExists: true)
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success},
                { assert snapshot(
                    workflow.out.fasta,
                    workflow.out.fasta_index,
                    workflow.out.bismark_index,
                    workflow.out.bwameth_index,
                    workflow.out.versions
                    ).match() }
            )
        }
    }

    test("Params: bwameth | generate fasta index | download bwameth index") {

        when {
            params {
                aligner = "bwameth"
            }
            workflow {
                """
                input[0] = file('https://github.com/nf-core/test-datasets/raw/methylseq/reference/genome.fa.gz', checkIfExists: true)
                input[1] = [] //fasta index
                input[2] = [] // bismark index
                input[3] = file('https://github.com/nf-core/test-datasets/raw/methylseq/reference/Bwameth_Index.tar.gz', checkIfExists: true)
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success},
                { assert snapshot(
                    workflow.out.fasta,
                    workflow.out.fasta_index,
                    workflow.out.bismark_index,
                    workflow.out.bwameth_index,
                    workflow.out.versions
                    ).match() }
            )
        }
    }

    test("Params: bwameth | generate fasta index | generate bwameth index") {

        when {
            params {
                aligner = "bwameth"
            }
            workflow {
                """
                input[0] = file('https://github.com/nf-core/test-datasets/raw/methylseq/reference/genome.fa.gz', checkIfExists: true)
                input[1] = [] // fasta_index
                input[2] = [] // bismark index
                input[3] = [] // bwameth index
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success},
                { assert snapshot(
                    workflow.out.fasta,
                    workflow.out.fasta_index,
                    workflow.out.bismark_index,
                    workflow.out.bwameth_index,
                    workflow.out.versions
                    ).match() }
            )
        }
    }

}
