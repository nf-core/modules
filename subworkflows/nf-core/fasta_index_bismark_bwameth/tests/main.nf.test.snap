{
    "Params: bismark | download bismark index (bowtie2)": {
        "content": [
            [
                [
                    {
                        
                    },
                    "genome.fa:md5,923a0a268ad29fee3c3437d00f9970de"
                ]
            ],
            [
                
            ],
            [
                [
                    {
                        
                    },
                    [
                        [
                            [
                                "BS_CT.1.bt2:md5,d53bef7d951c7d08e327944581c911eb",
                                "BS_CT.2.bt2:md5,752a03fc364115fc910e5a28ae154382",
                                "BS_CT.3.bt2:md5,b41e72eefad74e8ede411c78a5bd5dee",
                                "BS_CT.4.bt2:md5,02b3d1855a67fd9cbb4c411406a22fde",
                                "BS_CT.rev.1.bt2:md5,52ae603fff473b3b25eecd36c533edb9",
                                "BS_CT.rev.2.bt2:md5,0e290e05f7dae065cae54e29ed97afe5",
                                "genome_mfa.CT_conversion.fa:md5,4dd181b65f1f3549f4132c9a3759eee8"
                            ],
                            [
                                "BS_GA.1.bt2:md5,9aff51b1712758b891b0c427a988977f",
                                "BS_GA.2.bt2:md5,579241d95941538b2e75bbdb6cfaa73d",
                                "BS_GA.3.bt2:md5,b41e72eefad74e8ede411c78a5bd5dee",
                                "BS_GA.4.bt2:md5,6b64fa496ffa2af2af235e1c87f06a25",
                                "BS_GA.rev.1.bt2:md5,dabf2d6f80ba7e8fc96dfdc203f6bbfd",
                                "BS_GA.rev.2.bt2:md5,fac2f8b39b0e15dec7aa79a0ecf12b49",
                                "genome_mfa.GA_conversion.fa:md5,3dee732bc344fd620c3a7322f52c6a5f"
                            ]
                        ],
                        "genome.fa:md5,923a0a268ad29fee3c3437d00f9970de"
                    ]
                ]
            ],
            [
                
            ],
            [
                "versions.yml:md5,90950cf3155e6a77273de11ebb93a541",
                "versions.yml:md5,b0160273bd1b3b049a2748c02496c6d1"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.1"
        },
        "timestamp": "2024-11-26T16:24:46.519054694"
    },
    "Params: bwameth | generate fasta index | download bwameth index": {
        "content": [
            [
                [
                    {
                        
                    },
                    "genome.fa:md5,923a0a268ad29fee3c3437d00f9970de"
                ]
            ],
            [
                [
                    {
                        
                    },
                    "genome.fa.fai:md5,84f5b44c25877ac58b154706cc8bc451"
                ]
            ],
            [
                
            ],
            [
                [
                    {
                        
                    },
                    [
                        "genome.fa:md5,923a0a268ad29fee3c3437d00f9970de",
                        "genome.fa.bwameth.c2t:md5,e51d48ed28fa0c26e2f9c9f13d09403b",
                        "genome.fa.bwameth.c2t.amb:md5,010a242c6764efb30141868a45d698b3",
                        "genome.fa.bwameth.c2t.ann:md5,09b4db3d87a2d4dac9e10e807f377110",
                        "genome.fa.bwameth.c2t.bwt:md5,4646f8ae6bd523b7f4106bbd32eff95e",
                        "genome.fa.bwameth.c2t.pac:md5,a662d7add09698e25c8152fd2306ec66",
                        "genome.fa.bwameth.c2t.sa:md5,4a1e905b828396a1909669bc9c573d8d"
                    ]
                ]
            ],
            [
                "versions.yml:md5,90950cf3155e6a77273de11ebb93a541",
                "versions.yml:md5,b0160273bd1b3b049a2748c02496c6d1",
                "versions.yml:md5,ee0146f5d2942f9ff466bf275567515e"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.1"
        },
        "timestamp": "2024-11-26T16:26:40.995024346"
    },
    "Params: bwameth | generate fasta index | generate bwameth index": {
        "content": [
            [
                [
                    {
                        
                    },
                    "genome.fa:md5,923a0a268ad29fee3c3437d00f9970de"
                ]
            ],
            [
                [
                    {
                        
                    },
                    "genome.fa.fai:md5,84f5b44c25877ac58b154706cc8bc451"
                ]
            ],
            [
                
            ],
            [
                [
                    {
                        
                    },
                    [
                        "genome.fa.bwameth.c2t:md5,e51d48ed28fa0c26e2f9c9f13d09403b",
                        "genome.fa.bwameth.c2t.amb:md5,010a242c6764efb30141868a45d698b3",
                        "genome.fa.bwameth.c2t.ann:md5,09b4db3d87a2d4dac9e10e807f377110",
                        "genome.fa.bwameth.c2t.bwt:md5,4646f8ae6bd523b7f4106bbd32eff95e",
                        "genome.fa.bwameth.c2t.pac:md5,a662d7add09698e25c8152fd2306ec66",
                        "genome.fa.bwameth.c2t.sa:md5,4a1e905b828396a1909669bc9c573d8d"
                    ]
                ]
            ],
            [
                "versions.yml:md5,3a92f9ed1831ae3a68bc15886e7a95a1",
                "versions.yml:md5,90950cf3155e6a77273de11ebb93a541",
                "versions.yml:md5,ee0146f5d2942f9ff466bf275567515e"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.1"
        },
        "timestamp": "2024-11-26T16:27:06.257485123"
    },
    "Params: bismark_hisat | generate bismark index (hisat2)": {
        "content": [
            [
                [
                    {
                        
                    },
                    "genome.fa:md5,923a0a268ad29fee3c3437d00f9970de"
                ]
            ],
            [
                
            ],
            [
                [
                    {
                        
                    },
                    [
                        [
                            [
                                "BS_CT.1.ht2:md5,273c7686fc1dc7d93e36433592367f81",
                                "BS_CT.2.ht2:md5,752a03fc364115fc910e5a28ae154382",
                                "BS_CT.3.ht2:md5,b41e72eefad74e8ede411c78a5bd5dee",
                                "BS_CT.4.ht2:md5,02b3d1855a67fd9cbb4c411406a22fde",
                                "BS_CT.5.ht2:md5,d117de4c2c910a1ae7f0459c11ba6018",
                                "BS_CT.6.ht2:md5,4097a785a8d1ca92489d83b4c613a50b",
                                "BS_CT.7.ht2:md5,9013eccd91ad614d7893c739275a394f",
                                "BS_CT.8.ht2:md5,33cdeccccebe80329f1fdbee7f5874cb",
                                "genome_mfa.CT_conversion.fa:md5,4dd181b65f1f3549f4132c9a3759eee8"
                            ],
                            [
                                "BS_GA.1.ht2:md5,c2b02b7987f4ac0578332204678674cd",
                                "BS_GA.2.ht2:md5,579241d95941538b2e75bbdb6cfaa73d",
                                "BS_GA.3.ht2:md5,b41e72eefad74e8ede411c78a5bd5dee",
                                "BS_GA.4.ht2:md5,6b64fa496ffa2af2af235e1c87f06a25",
                                "BS_GA.5.ht2:md5,43aa1c78c6ec18c4165cf25159578fee",
                                "BS_GA.6.ht2:md5,2dc0f049d7d0f2a51822df8b195dce41",
                                "BS_GA.7.ht2:md5,9013eccd91ad614d7893c739275a394f",
                                "BS_GA.8.ht2:md5,33cdeccccebe80329f1fdbee7f5874cb",
                                "genome_mfa.GA_conversion.fa:md5,3dee732bc344fd620c3a7322f52c6a5f"
                            ]
                        ],
                        "genome.fa:md5,923a0a268ad29fee3c3437d00f9970de"
                    ]
                ]
            ],
            [
                
            ],
            [
                "versions.yml:md5,90950cf3155e6a77273de11ebb93a541",
                "versions.yml:md5,b0160273bd1b3b049a2748c02496c6d1"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.1"
        },
        "timestamp": "2024-11-26T16:26:00.163900219"
    },
    "Params: bismark_hisat | download bismark index (hisat2)": {
        "content": [
            [
                [
                    {
                        
                    },
                    "genome.fa:md5,923a0a268ad29fee3c3437d00f9970de"
                ]
            ],
            [
                
            ],
            [
                [
                    {
                        
                    },
                    [
                        [
                            [
                                "BS_CT.1.ht2:md5,273c7686fc1dc7d93e36433592367f81",
                                "BS_CT.2.ht2:md5,752a03fc364115fc910e5a28ae154382",
                                "BS_CT.3.ht2:md5,b41e72eefad74e8ede411c78a5bd5dee",
                                "BS_CT.4.ht2:md5,02b3d1855a67fd9cbb4c411406a22fde",
                                "BS_CT.5.ht2:md5,d117de4c2c910a1ae7f0459c11ba6018",
                                "BS_CT.6.ht2:md5,4097a785a8d1ca92489d83b4c613a50b",
                                "BS_CT.7.ht2:md5,9013eccd91ad614d7893c739275a394f",
                                "BS_CT.8.ht2:md5,33cdeccccebe80329f1fdbee7f5874cb",
                                "genome_mfa.CT_conversion.fa:md5,4dd181b65f1f3549f4132c9a3759eee8"
                            ],
                            [
                                "BS_GA.1.ht2:md5,c2b02b7987f4ac0578332204678674cd",
                                "BS_GA.2.ht2:md5,579241d95941538b2e75bbdb6cfaa73d",
                                "BS_GA.3.ht2:md5,b41e72eefad74e8ede411c78a5bd5dee",
                                "BS_GA.4.ht2:md5,6b64fa496ffa2af2af235e1c87f06a25",
                                "BS_GA.5.ht2:md5,43aa1c78c6ec18c4165cf25159578fee",
                                "BS_GA.6.ht2:md5,2dc0f049d7d0f2a51822df8b195dce41",
                                "BS_GA.7.ht2:md5,9013eccd91ad614d7893c739275a394f",
                                "BS_GA.8.ht2:md5,33cdeccccebe80329f1fdbee7f5874cb",
                                "genome_mfa.GA_conversion.fa:md5,3dee732bc344fd620c3a7322f52c6a5f"
                            ]
                        ],
                        "genome.fa:md5,923a0a268ad29fee3c3437d00f9970de"
                    ]
                ]
            ],
            [
                
            ],
            [
                "versions.yml:md5,90950cf3155e6a77273de11ebb93a541",
                "versions.yml:md5,b0160273bd1b3b049a2748c02496c6d1"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.1"
        },
        "timestamp": "2024-11-26T16:25:35.90378279"
    },
    "Params: bwameth | download fasta index | download bwameth index": {
        "content": [
            [
                [
                    {
                        
                    },
                    "genome.fa:md5,923a0a268ad29fee3c3437d00f9970de"
                ]
            ],
            [
                "/nf-core/test-datasets/raw/methylseq/reference/genome.fa.fai"
            ],
            [
                
            ],
            [
                [
                    {
                        
                    },
                    [
                        "genome.fa:md5,923a0a268ad29fee3c3437d00f9970de",
                        "genome.fa.bwameth.c2t:md5,e51d48ed28fa0c26e2f9c9f13d09403b",
                        "genome.fa.bwameth.c2t.amb:md5,010a242c6764efb30141868a45d698b3",
                        "genome.fa.bwameth.c2t.ann:md5,09b4db3d87a2d4dac9e10e807f377110",
                        "genome.fa.bwameth.c2t.bwt:md5,4646f8ae6bd523b7f4106bbd32eff95e",
                        "genome.fa.bwameth.c2t.pac:md5,a662d7add09698e25c8152fd2306ec66",
                        "genome.fa.bwameth.c2t.sa:md5,4a1e905b828396a1909669bc9c573d8d"
                    ]
                ]
            ],
            [
                "versions.yml:md5,90950cf3155e6a77273de11ebb93a541",
                "versions.yml:md5,b0160273bd1b3b049a2748c02496c6d1"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.1"
        },
        "timestamp": "2024-11-26T16:26:14.893224201"
    },
    "Params: bismark | generate bismark index (bowtie2)": {
        "content": [
            [
                [
                    {
                        
                    },
                    "genome.fa:md5,923a0a268ad29fee3c3437d00f9970de"
                ]
            ],
            [
                
            ],
            [
                [
                    {
                        
                    },
                    [
                        [
                            [
                                "BS_CT.1.bt2:md5,d53bef7d951c7d08e327944581c911eb",
                                "BS_CT.2.bt2:md5,752a03fc364115fc910e5a28ae154382",
                                "BS_CT.3.bt2:md5,b41e72eefad74e8ede411c78a5bd5dee",
                                "BS_CT.4.bt2:md5,02b3d1855a67fd9cbb4c411406a22fde",
                                "BS_CT.rev.1.bt2:md5,52ae603fff473b3b25eecd36c533edb9",
                                "BS_CT.rev.2.bt2:md5,0e290e05f7dae065cae54e29ed97afe5",
                                "genome_mfa.CT_conversion.fa:md5,4dd181b65f1f3549f4132c9a3759eee8"
                            ],
                            [
                                "BS_GA.1.bt2:md5,9aff51b1712758b891b0c427a988977f",
                                "BS_GA.2.bt2:md5,579241d95941538b2e75bbdb6cfaa73d",
                                "BS_GA.3.bt2:md5,b41e72eefad74e8ede411c78a5bd5dee",
                                "BS_GA.4.bt2:md5,6b64fa496ffa2af2af235e1c87f06a25",
                                "BS_GA.rev.1.bt2:md5,dabf2d6f80ba7e8fc96dfdc203f6bbfd",
                                "BS_GA.rev.2.bt2:md5,fac2f8b39b0e15dec7aa79a0ecf12b49",
                                "genome_mfa.GA_conversion.fa:md5,3dee732bc344fd620c3a7322f52c6a5f"
                            ]
                        ],
                        "genome.fa:md5,923a0a268ad29fee3c3437d00f9970de"
                    ]
                ]
            ],
            [
                
            ],
            [
                "versions.yml:md5,90950cf3155e6a77273de11ebb93a541",
                "versions.yml:md5,dc24c245324a3b8a8328f1ed6abd3f7c"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.1"
        },
        "timestamp": "2024-11-26T16:25:20.094964567"
    }
}